V1

ophirbucai opened the pull request.

ophirbucai committed:
Initial commit

Generated by create-expo-module 0.9.6.

ophirbucai committed:
feat: Implement files previewer on iOS

ophirbucai committed:
feat: Showcase examples

ophirbucai committed:
chore: Format with double quotes, remove unused import

ophirbucai committed:
chore: Format nl

ophirbucai committed:
chore: Add readme documentation and license

PierreJanineh committed:
Initial commit

ophirbucai committed:
chore: Remove example builds from Git

PierreJanineh commented on LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) Pierre Janineh and Ophir Bucai
```
<3

PierreJanineh commented:
`android` directory needs to be removed.
Android is not yet supported.

PierreJanineh reviewed (approved):
`android` directory needs to be removed.
Android is not yet supported.

ophirbucai committed:
chore: Add native build directories to .gitignore

ophirbucai committed:
refactor: Use requireOptionalNativeModule for platform compatability, link to upvote issue

ophirbucai committed:
refactor: Update module on Android to include a placeholder for future implementation

ophirbucai committed:
refactor: Rename module to the library's name

ophirbucai committed:
refactor: Revert renaming module declaration

ophirbucai committed:
feat: Add support for file name parameter, with a fallback of `lastPathComponent`

ophirbucai committed:
upgrade to version 0.1.1

ophirbucai committed:
feat: Update examples with descriptive file names

ophirbucai commented:
Got some feedback that we should take into consideration:

1. Options should be an object, that way the parameters are named when passing them into the `preview` function. I think it makes sense, because we may want to support other parameters, and fileName being optional would feel quite strange (think of this scenario `preview(fileUrl, undefined, true, 5)`, it would be unclear what those parameters represent and their order).
2. I'm currently exporting the native module using requireOptionalNativeModule, which attempts to resolve the module, if not found, it returns null. So when using the default export, it must be used with optional chaining. Not the best experience. I need to see if we can generate just a no-op class instead, maybe use expo's ability to differentiate between different file exports (`fileName.ts` | `fileName.ios.ts` are picked when importing, based on the platform).

PierreJanineh committed:
feat: add PreviewOptions support for structured object previews

- Updated the `preview` function to accept a `PreviewOptions` object, allowing for structured parameter input in TypeScript.
- Improved type safety and readability by integrating the `PreviewOptions` interface.
- Updated the module declaration for better platform compatibility documentation and usage examples.

BREAKING CHANGE: Previously, `preview` accepted individual string arguments, but now it requires a structured object (`PreviewOptions`).

ophirbucai committed:
chore: Remove redundant comment, organize imports

ophirbucai committed:
chore: Update the readme usage example to align with the new API

ophirbucai committed:
chore: Remove unneeded nl

ophirbucai committed:
chore: Remove unneeded comments, format code

ophirbucai committed:
feat: Enhance preview method with error handling and custom messages

ophirbucai committed:
feat: Export preview method directly

ophirbucai committed:
chore: Fix formatting

ophirbucai committed:
chore: Improve JSDoc
